lib/Frontend/CompilerInvocation.cpp

```diff
@@ -886,6 +899,41 @@ static bool ParseSearchPathArgs(SearchPathOptions &Opts,
   return false;
 }
 
+static void ValidateLocalizationArgs(ArgList &Args, DiagnosticEngine &Diags) {
```

Comment:
Should this accept `DiagnosticOptions &` instead here? Because that would be either set to what user provided or default value. I don't see much reason to re-fetch locale/path...